src/main/java/com/squareup/javapoet/CodeBlock.java

```diff
@@ -112,9 +118,64 @@ public Builder toBuilder() {
     private Builder() {
     }
 
+    /**
+     * Adds code using named arguments.
+     *
+     * <p> Named arguments specify their name after the '$' followed by : and the corresponding
+     * type character. For example, to refer to the class {@link java.lang.String} with the argument
+     * name 'text' use a format string containing {@code $text:S} and include the key 'text' with
```

Comment:
this is a really distracting example! Maybe simpler to give an example that doesn’t require a type conversion?